File: XUCore/localization/NSWindowLocalizationSupport.m
// 
// NSWindowLocalizationSupport.m
// 
// Created by Charlie Monroe
// 
// Copyright (c) 2010-13 Fuel Collective, LLC. All rights reserved.
// 


#import "NSWindowLocalizationSupport.h"

#ifndef FCLocalizedString
	#import "FCLocalizationSupport.h"
#endif

@implementation NSWindow (NSWindowLocalizationSupport)
-(void)localizeWindow{
	[self setTitle:FCLocalizedString([self title])];
	[[self contentView] localizeView];
	
	NSToolbar *toolbar = [self toolbar];
	for (NSToolbarItem *item in [toolbar items]){
		[item setLabel:FCLocalizedString([item label])];
	}
}
@end

File: XUCore/localization/NSWindowLocalizationSupport.swift
//
//  NSWindowLocalizationSupport.swift
//  XUCore
//
//  Created by Charlie Monroe on 11/7/15.
//  Copyright © 2015 Charlie Monroe Software. All rights reserved.
//

import Foundation

public extension NSWindow {
	
	public func localizeWindow(_ bundle: Bundle = XUMainBundle) {
		self.title = XULocalizedString(self.title, inBundle: bundle)
		
		self.contentView?.localizeView(bundle)
		
		if let toolbar = self.toolbar {
			for item in toolbar.items {
				item.label = XULocalizedString(item.label, inBundle: bundle)
			}
		}
	}
	
}
